Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -221,14 +221,16 @@ public class HTTPClient {
     }
 
     private func execute<Delegate: HTTPClientResponseDelegate>(request: Request,
+                                                               redirectLimit: Configuration.RedirectLimit.Next? = nil,
```

Comment:
another option to consider is to make the limit part of the request object (internally)

Reply:
I did try that option as well, but current implementation seemed a bit better, would you like me to commit alternative version so we can compare?

Reply:
sure if you are up to it. feels better encapsulated on the surface

Reply:
I've updated the PR with redirect state in request